### ISSUE TITLE
Generics broken when using a 'prototype' in typecheck

### ISSUE THREAD

**Issue by 6finger** [Microsoft/TypeScript#9738]
**TypeScript Version:**  1.8.0 / nightly (2.0.0-dev.201xxxxx)
The one running on http://www.typescriptlang.org/play/index.html on 14.7.2016

**Code**

```ts
// the last example from http://www.typescriptlang.org/docs/handbook/generics.html
class BeeKeeper {
    hasMask: boolean;
}

class ZooKeeper {
    nametag: string;
}

class Animal {
    numLegs: number;
}

class Bee extends Animal {
    keeper: BeeKeeper;
}

class Lion extends Animal {
    keeper: ZooKeeper;
}

function findKeeper<A extends Animal, K> (a: {new(): A;
    prototype: {keeper: K}}): K {

    return a.prototype.keeper;
}

findKeeper(Lion).nametag;  // typechecks!

```

**Expected behavior:**
typecheck should report an error because **findKeeper** returns a **ZooKeeper** which is a constructor and **nametag** is not a property on the constructor. It is on the instance.

**Actual behavior:** 
typecheck does not fail

**Comment by RyanCavanaugh** [comment]
I don't understand why you think this should be an error. The `.keeper` property refers to a `ZooKeeper` *instance*. It's not the same as `typeof ZooKeeper`, the type of the constructor function.

**Comment by 6finger** [comment]
Thanks for quick response. I went through it again and think you're right. I got confused when I tried the example in the playground.
This is what I tried to add at the end of the example:
```
try {
	alert(findKeeper(Lion)); // undefined
	alert(findKeeper(Lion).nametag);  // typechecks but fails on runtime
} catch (e) {
	alert(e);
}
```
How does the instance of `ZooKeeper` that should be the value of `Lion.prototype.keeper` gets created?
I assume I'm responsible for creating it and Typescript just checks that the property is declared to be there, right?

**Comment by RyanCavanaugh** [comment]
> I assume I'm responsible for creating it and Typescript just checks that the property is declared to be there, right?

Exactly

**Comment by 6finger** [comment]
Thanks for your help :)
